src/java/org/apache/cassandra/config/ConfigFields.java

```diff
@@ -0,0 +1,49 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.cassandra.config;
+
+/**
+ * This class is created by {@link org.apache.cassandra.utils.ConfigFieldsGenerateUtil} based on provided
+ * the {@link org.apache.cassandra.config.Config} class. It contains non-private non-static {@code Cofig}'s fields
+ * marked with the {@link org.apache.cassandra.config.Mutable} annotation to expose them to public APIs.
+
+ * @see org.apache.cassandra.config.Mutable
+ * @see org.apache.cassandra.config.Config
+ */
+public class ConfigFields
```

Comment:
how is this class going to be updated when something changes? I see we have a generator for that but maybe we should hook the execution of this generator to the build process so we would generate this automatically? (or we would fail the build if the generated class is different from what is committed so people have to be aware of the fact they have to regenerate it themselves)

Reply:
The problem with having this class generated at the build is the number of classes the Config depends on: 43. We have to compile them all first (and their dependencies accordingly), which makes the build process itself more complicated than it needs to be. Using an external library to achieve the same result also seems like over-engineering.

We've argued with David about the `ConfigFields` class itself (he doesn't like it at all), but I don't see any other good options here, as a Config field name can be used several times in the codebase mostly in the `DatabaseDescription` class. If we want to rename a config field we either rename it once in the `ConfigFields` class followed by the link to the class in javadoc, or we have to search for the name in all the sources.

So the `ConfigFieldsGenerateUtil` is just used as a convenient way to create these constants until they all have the @Mutable annotation (or none if it is not needed for a field) and then remove it from the source code. Thoughts?

Reply:
I dont know ... feels like a hack. Is that really necessary? Why not to put these constants directly in DatabaseDescriptor and be done with it?

Reply:
I think this is possible, or just move the constants under the inner `Fields` class in the `Config` itself. Both are ok.

Reply:
Yeah something like that, up to you. The goal is to not generate and have code around that.

Reply:
Moved to the `Config`, the name updated accordingdly.